src/Tribe/Repositories/Event.php

```diff
@@ -216,6 +190,30 @@ public function __construct() {
 		$this->add_simple_tax_schema_entry( $tribe_events_category . '_not_in', $tribe_events_category, 'term_not_in' );
 		$this->add_simple_tax_schema_entry( 'tag', 'post_tag' );
 		$this->add_simple_tax_schema_entry( 'tag_not_in', 'post_tag', 'term_not_in' );
+
+		add_action('pre_get_posts', [ $this, 'ensure_event_post_types' ], 50 );
+	}
+
+	/**
+	 * Ensures that if something hooks into `pre_get_posts` and alters the query post type
+	 * the `tribe_events` post type is preserved so event searches don't break.
+	 *
+	 * @since TBD
+	 *
+	 * @param  WP_Query $query
+	 *
+	 * @return  WP_Query modified query
+	 */
+	public function ensure_event_post_types( $query ) {
+		if ( ! $query->is_search ) {
+			return $query;
+		}
+
+		if ( 'tribe_events' === $query->query['post_type'] ) {
```

Comment:
use `Tribe__Events__Main::POSTTYPE` but this also needs to happen only there queries executed with the ORM